plugins/grafana-sankey-plugin/src/SankeyPanel.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { PanelProps } from '@grafana/data';
+import { SankeyOptions } from 'types';
+import Chart from 'react-google-charts';
+
+interface Props extends PanelProps<SankeyOptions> {}
+
+export const SankeyPanel: React.FC<Props> = ({ options, data, width, height }) => {
+  let result = [
+    ['From', 'To', 'Bytes'],
+    ['Source N/A', 'Destination N/A', 1],
+  ];
+  let sources = data.series
+    .map((series) => series.fields.find((field) => field.name === 'source'))
+    .map((field) => {
+      let record = field?.values as any;
+      return record?.buffer;
+    })[0];
+  if (sources !== undefined) {
+    let destinations = data.series
+      .map((series) => series.fields.find((field) => field.name === 'destination'))
+      .map((field) => {
+        let record = field?.values as any;
+        return record?.buffer;
+      })[0];
+    let destinationIPs = data.series
+      .map((series) => series.fields.find((field) => field.name === 'destinationIP'))
+      .map((field) => {
+        let record = field?.values as any;
+        return record?.buffer;
+      })[0];
+    let bytes = data.series
+      .map((series) => series.fields.find((field) => field.name === 'bytes'))
+      .map((field) => {
+        let record = field?.values as any;
+        return record?.buffer;
+      })[0];
+    let n = sources.length;
+    for (let i = 0; i < n; i++) {
+      let record = [];
+      let source = sources[i];
+      let destination = destinations[i];
+      record.push(source);
+      if (destination === '') {
+        record.push(destinationIPs[i]);
+      } else {
+        if (source === destination) {
+          destination = destination + ' ';
```

Comment:
just curious about why a space is appended here. Do we need to distinguish it from source otherwise it won't get appended to the record?

Reply:
We found that if source and destination is the same, Google chart will not be rendered due to cycle detected. https://stackoverflow.com/questions/64164313/google-chart-sankey-cycle-found-in-rows-error
Other workaround we thought is to add `source` and `destination` prefix, but it seems redundant.

Reply:
Thanks for the clarification. If an extra space does the trick, that's great. Perhaps I would just suggest to add a comment in the code with the same clarification in case someone else has the same question in 2 years' time :)

Reply:
Sure. Updated. Thanks!